File: fixtures/typescript/typescript-simple/src/in-memory-user-repository.ts
import type { User, UserId } from './user'
import { UserStatus } from './user'
import type { UserRepository } from './user-repository'

export class InMemoryUserRepository implements UserRepository {
  private readonly users = new Map<UserId, User>([
    ['42', { id: '42', name: 'Ada', status: UserStatus.ACTIVE }],
  ])

  findById(id: UserId): User | undefined {
    return this.users.get(id)
  }
}

File: fixtures/java/java-simple/src/main/java/com/minos/fixture/InMemoryUserRepository.java
package com.minos.fixture;

import java.util.Map;
import java.util.Optional;

public final class InMemoryUserRepository implements UserRepository {

    private final Map<String, User> users = Map.of(
            "42", new User("42", "Ada")
    );

    @Override
    public Optional<User> findById(String id) {
        return Optional.ofNullable(users.get(id));
    }
}
